assets/prototype/domain/eventEditor/containers/queries/useCacheRehydration.js

```diff
@@ -0,0 +1,84 @@
+import { useEffect } from '@wordpress/element';
+import { useApolloClient } from '@apollo/react-hooks';
+import useCacheRehydrationData from './useCacheRehydrationData';
+import useRelations from '../../../../infrastructure/services/relations/useRelations';
+import useStatus from '../../../../infrastructure/services/status/useStatus';
+import useEventId from './useEventId';
+import { queries } from './';
+
+const { GET_TICKETS, GET_DATETIMES, GET_PRICE_TYPES, GET_PRICES } = queries;
+
+const useCacheRehydration = () => {
```

Comment:
@manzoorwanijk this isn't exactly the approach I saw here:
https://www.apollographql.com/docs/react/performance/server-side-rendering/#store-rehydration

Reply:
my understanding is that this is updating cache on `EventEditor` component mount

Reply:
I want to understand why we didn't go through this path: `cache: new InMemoryCache().restore(window.__APOLLO_STATE__)` ?

Reply:
The reason for that is we don't use Apollo Server on the backend. So we need to write to the cache using `client.writeQuery` as mentioned [here](https://www.apollographql.com/docs/react/data/local-state/#initializing-the-cache).